### DISCUSSION TITLE
Specification: how nested inline tables should be treated?

### DISCUSSION THREAD

**Discussion by orchestr7** [toml-lang/toml#879] [Q&A] [answered]
Hi, I am implementing [toml deserializer](https://github.com/akuleshov7/ktoml) for Kotlin (ktoml) and not able to understand how to handle following TOMLs:
```
table2 = { table2."akuleshov7.com" = { name = "test" } }

VS

table2 = { "akuleshov7.com" = { name = "test" } }
```

In TOML standard it is specified, that nested tables should have the prefix of it's parent (`table2`) :
```
[table2]
[table2."akuleshov7.com"]
```

Is the same applied to inline tables and they should also have the prefix?

**Comment by marzer** [comment]
> Is the same applied to inline tables and they should also have the prefix?

Yes. There's no difference between the two syntaxes for deserialisation purposes. The consuming application should see the same data either way.

**Reply by orchestr7** [reply]
> Yes. There's no difference between the two syntaxes for deserialisation purposes. The consuming application should see the same data either way.


@marzer thank you for your reply! Could you please translate my above examples with **inline** tables into the **simple** table representation? Right now I cannot understand if I have to use **full prefixes** for nested tables in inline structures.

```
table2 = { table2."akuleshov7.com" = { name = "test" } }

VS

table2 = { "akuleshov7.com" = { name = "test" } }
```

**Reply by marzer** [reply]
Ah, looks like I might have misunderstood your question. The prefix doesn't need to appear in the TOML. I meant that from the perspective of the consuming application, both forms should have the same structure, as if the inline table version had the parent prefixes implicitly.

**Reply by marzer** [reply]
So, in your example above:
```toml
table2 = { table2."akuleshov7.com" = { name = "test" } }
```
is the same as
```toml
[table2.table2."akuleshov7.com"]
name = "test"
```
and
```toml
table2 = { "akuleshov7.com" = { name = "test" } }
```
is the same as
```toml
[table2."akuleshov7.com"]
name = "test"
```

**Comment by orchestr7** [comment, marked as answer]
@marzer got it, thanks. So the right answer is the following: prefixes are not needed for nested tables in inline structures. 

This code:
```
table2 = { table2."akuleshov7.com" = { name = "test" }, i = 1 }

VS

table2 = { "akuleshov7.com" = { name = "test" }, i = 1 }
```

is equal to the following one:
```
[table2]
    i = 1
[table2.table2."akuleshov7.com"]
    name = "test"

VS 

[table2]
   i = 1
[table2."akuleshov7.com"]
    name = "test"
```

**Reply by marzer** [reply]
Yeh. Using a tool like [this one](https://www.convertsimple.com/convert-toml-to-json/) to see the equivalent structures in JSON can help with disambiguation in cases like this.
